crates/astria-sequencer-client/src/tests/http.rs

```diff
@@ -120,57 +129,69 @@ fn create_signed_transaction() -> transaction::Signed {
     let alice_keypair = SigningKey::from(alice_secret_bytes);
 
     let actions = vec![transaction::Action::TransferAction(Transfer::new(
-        Address::try_from_str(BOB_ADDRESS).unwrap(),
+        Address::try_from(BOB_ADDRESS.as_slice()).unwrap(),
         Balance::from(333_333),
     ))];
     let tx = transaction::Unsigned::new_with_actions(Nonce::from(1), actions);
     tx.into_signed(&alice_keypair)
 }
 
 #[tokio::test]
-async fn get_nonce_and_balance() {
-    let server = MockTendermintServer::new().await;
-
-    let nonce_response = Nonce::from(0);
-    server
-        .register_abci_query_response(
-            "accounts/nonce",
-            &query::Response::NonceResponse(nonce_response),
-        )
-        .await;
-
-    let balance_response = Balance::from(10_u128.pow(18));
-    server
-        .register_abci_query_response(
-            "accounts/balance",
-            &query::Response::BalanceResponse(balance_response),
-        )
-        .await;
-
-    let client = HttpClient::new(server.address().as_str()).unwrap();
-    let address = Address::try_from_str(ALICE_ADDRESS).unwrap();
-    let nonce = client.get_nonce(&address, None).await.unwrap();
-    assert_eq!(nonce, nonce_response);
-    let balance = client.get_balance(&address, None).await.unwrap();
-    assert_eq!(balance, balance_response);
+async fn get_latest_nonce() {
+    use proto::sequencer::v1alpha1::NonceResponse;
+    let MockSequencer {
+        server,
+        client,
+    } = MockSequencer::start().await;
+
+    let expected_response = NonceResponse {
+        account: ALICE_ADDRESS.to_vec(),
+        height: 10,
+        nonce: 1,
+    };
+    let _guard =
```

Comment:
what's the purpose of declaring `_guard`?

Reply:
It's to provide a cleaner way of tracking which mock failed. Since `register_abci_query_response` uses `Mock::mount_as_scoped`, the `expect(1)` condition is checked when `_guard` goes out of scope. If it were mounted using just `Mock::mount`, the condition would be checked only once the entire server goes out of scope.

So doing it this way you will know that the condition associated with this `_guard` is the culprit.

The underscore is just because `_guard`'s entire purpose is to be checked on drop, and rust's dead code analysis would complain that `guard` is never used.